internal/model/virtualmtime.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2014 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package model
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/syncthing/syncthing/internal/db"
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+type virtualMtimeRepo struct {
+	ns *db.NamespacedKV
+}
+
+type virtualMtimeRecord struct {
+	DiskMtime   time.Time
+	ActualMtime time.Time
+}
+
+func newVirtualMtimeRepo(ldb *leveldb.DB, folder string) *virtualMtimeRepo {
+	prefix := string(db.KeyTypeVirtualMtime) + folder
+
+	return &virtualMtimeRepo{
+		ns: db.NewNamespacedKV(ldb, prefix),
+	}
+}
+
+func (r *virtualMtimeRepo) UpdateMtime(path string, diskMtime, actualMtime time.Time) {
+	// TODO need error handling?
```

Comment:
Looks like currently it's guaranteed (although not specified) to return a 14 byte slice: http://golang.org/src/time/time.go?s=24443:24488#L832. I'd probably go with that assumption and add a panic for when it's not true any more and we'll notice...